arviz/data/datasets.py

```diff
@@ -0,0 +1,183 @@
+(gamma) for contextual effects of the uranium per household.
+
+See Gelman and Hill (2006) for details on the example, or
+https://docs.pymc.io/notebooks/multilevel_modeling.html#Correlations-among-levels
+by Chris Fonnesbeck for details on this implementation.
+""",
+    )
+}
+
+
+def get_data_home(data_home=None):
+    """Return the path of the arviz data dir.
+
+    This folder is used by some dataset loaders to avoid downloading the
+    data several times.
+
+    By default the data dir is set to a folder named 'arviz_data' in the
+    user home folder.
+
+    Alternatively, it can be set by the 'ARVIZ_DATA' environment
+    variable or programmatically by giving an explicit folder path. The '~'
+    symbol is expanded to the user home folder.
+
+    If the folder does not already exist, it is automatically created.
+
+    Parameters
+    ----------
+    data_home : str | None
+        The path to arviz data dir.
+    """
+    if data_home is None:
+        data_home = os.environ.get("ARVIZ_DATA", os.path.join("~", "arviz_data"))
+    data_home = os.path.expanduser(data_home)
+    if not os.path.exists(data_home):
+        os.makedirs(data_home)
+    return data_home
+
+
+def clear_data_home(data_home=None):
+    """Delete all the content of the data home cache.
+
+    Parameters
+    ----------
+    data_home : str | None
+        The path to arviz data dir.
+    """
+    data_home = get_data_home(data_home)
+    shutil.rmtree(data_home)
+
+
+def _sha256(path):
+    """Calculate the sha256 hash of the file at path."""
+    sha256hash = hashlib.sha256()
+    chunk_size = 8192
+    with open(path, "rb") as buff:
+        while True:
+            buffer = buff.read(chunk_size)
+            if not buffer:
+                break
+            sha256hash.update(buffer)
+    return sha256hash.hexdigest()
+
+
+def load_arviz_data(dataset=None, data_home=None):
+    """Load a local or remote pre-made dataset.
+
+    Run with no parameters to get a list of all available models.
+
+    The directory to save to can also be set with the environement
+    variable `ARVIZ_HOME`. The checksum of the dataset is checked against a
+    hardcoded value to watch for data corruption.
+
+    Run `az.clear_data_home` to clear the data directory.
+
+    Parameters
+    ----------
+    dataset : str
+        Name of dataset to load.
+
+    data_home : str, optional
+        Where to save remote datasets
+
+    Returns
+    -------
+    xarray.Dataset
+    """
+    if dataset in LOCAL_DATASETS:
+        resource = LOCAL_DATASETS[dataset]
+        return load_data(resource.filename)
+
+    elif dataset in REMOTE_DATASETS:
+        remote = REMOTE_DATASETS[dataset]
+        home_dir = get_data_home(data_home=data_home)
+        file_path = os.path.join(home_dir, remote.filename)
+        if not os.path.exists(file_path):
+            urlretrieve(remote.url, file_path)
+        checksum = _sha256(file_path)
+        if remote.checksum != checksum:
```

Comment:
Would it be beneficial to provide some instructions on how to delete the offending file to allow for a redownload? For newer users they might just keep retrying this command get and get frustrated that the checksum fails every time

Reply:
Good idea!